### PR TITLE
feat: adds github workflow to create an ironic esp image

### PR DESCRIPTION
**Pull request by nicholaskuechler**
https://docs.openstack.org/ironic/latest/install/configure-esp.html
https://docs.openstack.org/ironic/latest/install/configure-pxe.html#uefi-pxe-grub

Screenshot 2024-11-07 at 3 13 24 PM

### CODE REVIEW

**Review by cardoe — CHANGES_REQUESTED** [review]
Add this to the regular images workflow so it's all released together.

File: .github/workflows/build-ironic-images.yaml [hunk 1]
```diff
@@ -86,3 +137,13 @@ jobs:
...
           files: ironic-images/upload/*
         if: ${{ github.ref == 'refs/heads/main' }}
+
+  cleanup:
+    runs-on: ubuntu-latest
+    needs: build-tenant-images
+
+    steps:
+      - name: Clean up artifacts
+        uses: geekyeggo/delete-artifact@v5
```
Comment (cardoe) [inline .github/workflows/build-ironic-images.yaml:147]: Is this just so we don't have a ton of artifacts?

Comment (nicholaskuechler) [reply .github/workflows/build-ironic-images.yaml:147]: Yep exactly. I think the default artifact lifetime is 90 days.

Comment (cardoe) [reply .github/workflows/build-ironic-images.yaml:147]: We can set the retention time to 1 day but I won't block this going in.

**Review by cardoe — APPROVED** [review]
So this proves what I was saying on the original change here that the release needs to be in a final step as one upload.

**Pull request merged**